resources/providers/azurelib/inventory/security_provider.go

```diff
@@ -0,0 +1,153 @@
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package inventory
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/security/armsecurity"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/samber/lo"
+
+	"github.com/elastic/cloudbeat/resources/utils/maps"
+	"github.com/elastic/cloudbeat/resources/utils/pointers"
+)
+
+const listSecurityContactsURI = "/subscriptions/%s/providers/Microsoft.Security/securityContacts"
+
+// azureSecurityContactsClient is needed till this issue -
+// https://github.com/Azure/azure-sdk-for-go/issues/19740 is fixed.
+// Implementation code is similar to armsecurity.ContactsClient but it attempts to decode -
+// the List response to two possible structs.
+type azureSecurityContactsClient struct {
+	armClient *arm.Client
+}
+
+// ListSecurityContacts implements the Security Contacts List Azure REST API call.
+// https://learn.microsoft.com/en-us/rest/api/defenderforcloud/security-contacts/list
+func (c *azureSecurityContactsClient) ListSecurityContacts(ctx context.Context, subID string) (armsecurity.ContactsClientListResponse, error) {
+	req, err := c.listSecurityContactsRequest(ctx, c.armClient.Endpoint(), subID)
+	if err != nil {
+		return armsecurity.ContactsClientListResponse{}, err
+	}
+
+	httpResp, err := c.armClient.Pipeline().Do(req)
+	if err != nil {
+		return armsecurity.ContactsClientListResponse{}, err
+	}
+
+	if !runtime.HasStatusCode(httpResp, http.StatusOK) {
+		err = runtime.NewResponseError(httpResp)
+		return armsecurity.ContactsClientListResponse{}, err
+	}
+
+	return c.listSecurityContactsResponseDecode(httpResp)
+}
+
+// listCreateRequest creates the List request.
+func (*azureSecurityContactsClient) listSecurityContactsRequest(ctx context.Context, endpoint, subID string) (*policy.Request, error) {
+	if subID == "" {
+		return nil, errors.New("parameter subscription id cannot be empty")
+	}
+
+	urlPath := fmt.Sprintf(listSecurityContactsURI, url.PathEscape(subID))
+
+	req, err := runtime.NewRequest(ctx, http.MethodGet, runtime.JoinPaths(endpoint, urlPath))
+	if err != nil {
+		return nil, err
+	}
+
+	reqQP := req.Raw().URL.Query()
+	reqQP.Set("api-version", "2020-01-01-preview")
```

Comment:
The whole azure thing with api versions 🙄 

Maybe a good idea to create a constant for it just for easier findability? (totally optional ofc)

Reply:
This "low level" client should be temporary (til the issue https://github.com/Azure/azure-sdk-for-go/issues/19740 is fixed).  I am not sure we will use these values anywhere else. 

If we will we could extract them, sure.